pithos/pithos.py

```diff
@@ -1204,7 +1204,15 @@ def song_text(self, song):
         album = html.escape(song.album)
         msg = []
         if song is self.current_song:
-            song.position = self.query_position()
+            # Avoid race condition where query_position() sometimes points us
+            # permanently to an invalid segment if we are called whilst
+            # buffering. Most reliable semaphore for buffering state is
+            # buffering_timer_id, cleared after buffering (see on_gst_buffering())
+            if self.buffering_timer_id == 0:
+                song.position = self.query_position()
```

Comment:
I'm not sure what an "invalid segment" is, but can you not change `query_position()` to never return that?